### PR TITLE
feat: oauth add passkey after signup (popup flow)

### PR DESCRIPTION
**Pull request by linnall**
# Pull Request Checklist

- [ ] Did you add new tests and confirm existing tests pass? (`yarn test`)
- [ ] Did you update relevant docs? (docs are found in the `site` folder, and guidelines for updating/adding docs can be found in the [contribution guide](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md))
- [ ] Do your commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] Does your PR title also follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] If you have a breaking change, is it [correctly reflected in your commit message](https://www.conventionalcommits.org/en/v1.0.0/#examples)? (e.g. `feat!: breaking change`)
- [ ] Did you run lint (`yarn lint:check`) and fix any issues? (`yarn lint:write`)
- [ ] Did you follow the [contribution guidelines](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md)?

---

## PR-Codex overview
This PR introduces a new user signup event in the `AlchemySigner` and updates various components to handle this event, alongside some refactoring in the `useAuthModal` and related hooks.

### Detailed summary
- Added `newUserSignup` event to `AlchemySignerClientEvents`.
- Emitted `newUserSignup` on successful user signup in `AlchemySignerWebClient`.
- Refactored `closeAuthModal` in `useAuthModal` to improve readability.
- Updated `CompletingOAuth` to simplify auth step setting.
- Adjusted `AuthModal` to handle signup logic with `useNewUserSignup`.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: account-kit/signer/src/client/types.ts [hunk 1]
```diff
@@ -142,6 +142,7 @@ export type AuthenticatingEventMetadata = {
 
 export type AlchemySignerClientEvents = {
   connected(user: User): void;
+  newUserSignupClient(): void;
```
Comment (moldy530) [inline account-kit/signer/src/client/types.ts:145]: this is super nitpicky... but can we drop the `Client` from the event name? the origin is implied when you subscribe already

**Review by moldy530 — COMMENTED** [review]
do we need to update the ui-demo app to default back to using pop-up now?

File: account-kit/react/src/components/auth/modal.tsx [hunk 2]
```diff
@@ -16,19 +16,20 @@ export const AuthModal = () => {
...
       openAuthModal();
+
       setAuthStep({
         type: "passkey_create",
       });
     }
-  }, [addPasskeyOnSignup, isOpen, openAuthModal, setAuthStep]);
+  }, [addPasskeyOnSignup, openAuthModal, setAuthStep]);
 
-  useNewUserSignup(handleSignup, isConnected);
```
Comment (moldy530) [inline account-kit/react/src/components/auth/modal.tsx:31]: this solves a race condition where an on success handler would set the step to complete and close the modal that was just opened. This makes sure that all handlers complete we're done with the setup flow prior to add passkey

File: account-kit/react/src/hooks/useAuthModal.ts [hunk 3]
```diff
@@ -33,7 +33,10 @@ export const useAuthModal = () => {
   );
 
   const openAuthModal = useCallback(() => setModalOpen(true), [setModalOpen]);
-  const closeAuthModal = useCallback(() => setModalOpen(false), [setModalOpen]);
+  const closeAuthModal = useCallback(() => {
+    console.trace("closing modal");
```
Comment (moldy530) [inline account-kit/react/src/hooks/useAuthModal.ts:37]:
```suggestion
```

File: account-kit/react/src/components/auth/modal.tsx (hunk 2 above)
Comment (linnall) [reply account-kit/react/src/components/auth/modal.tsx:31]: yay thanks for looking into this!

**Comment by linnall** [comment]
> do we need to update the ui-demo app to default back to using pop-up now?

done in #1140 before merging ([here](https://github.com/alchemyplatform/aa-sdk/pull/1140/commits/8d10d7009642905e5a184c6d79702123baeefcf0) and [here](https://github.com/alchemyplatform/aa-sdk/pull/1140/commits/d952810f1a6da0a3e611b9b52221a8e782a3dd7b))

**Pull request merged**